File: NBA_taehak/2023_03/week_4th/11726.py
# [BOJ] 11726. 2xn 타일링
# 소요 시간 : 00분
size = int(input())
result = [0, 1, 2]

for idx in range(3, size + 1):
    result.append(result[-1] + result[-2])
print(result[size]%10007)

File: NBA_deokgyun/2023_03/week_4th/11726.java
import java.util.Scanner;

// [BOJ] 11726. 2xn 타일링
// 소요 시간 : 00분
class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        long[] memo = new long[n + 1];
        memo[0] = 1;
        memo[1] = 1;
        for (int i = 2; i <= n; i++){
            memo[i] = (memo[i - 1] + memo[i - 2]) % 10007;
        }
        long result = memo[n];
        System.out.println(result);
    }
}